### DISCUSSION TITLE
Auto-Publish scheduled drafts

### DISCUSSION THREAD

**Discussion by deeekay** [craftcms/cms#14818] [Ideas]
In Craft we can use drafts to create and maintain future versions of entries. These drafts can also be scheduled (just as active entries can). In addition to that, it would be very helpful for a lot of our customers if drafts could be "pre-published" (i.e. validated) and automatically published according to their scheduled date and time.

As a reference, we observe this in some other CMS like, for example, Kentico (see: https://docs.kentico.com/13/managing-website-content/working-with-pages/scheduling-pages-for-publishing#scheduling-pages-that-use-workflowversioning).

The mentioned feature could also be beneficial for entries of type Single in order to schedule future versions of, for example, the homepage of a website.

Please let us know if and how this could (or could not) fit into the Craft CMS plans and roadmap.

Best regards,
Daniel

**Comment by brandonkelly** [comment]
We want to add support for auto-published drafts, but it will require some sort of job scheduler to manage it. We’re hoping we can build that on AWS EventBridge rather than relying on a PHP-based job scheduler, which would be more costly.

**Comment by deeekay** [comment]
@brandonkelly Thanks for the prompt reply. That sounds promising 🙏 do you already have an ETA for such (a) feature(s)?

In my opinion, it could also be an option to make the job scheduler the user's responsibility, e.g. by setting up a cron job for a craft console command.

**Reply by brandonkelly** [reply]
No ETA, sorry. But yes, providing the scheduler will need to be something that’s mostly provided by the developer.

**Comment by patrikalienus** [comment]
We have been relying on Publisher X for this specific functionality. However it is no longer working and the developer seem to have abandoned the plugin, at least for v4.x, which is what we're currently on (trying to get the codebase ready for v5 obviously)

**Comment by Aku-benj** [comment]
We have some clients with this need too. 
Would be awesome to also be able to schedule the re-publishing/re-enabling of disabled contents.
